### DISCUSSION TITLE
How to fix the type issues with pure action returned actions

### DISCUSSION THREAD

**Discussion by zhangjialu** [statelyai/xstate#3842] [Q&A]
![1676733085447](https://user-images.githubusercontent.com/20961154/219873310-6d409051-b912-4c0a-bd42-e26932be6bb2.png)

It seems that typegen doesn't know the actions returned by pure action

**Comment by davidkpiano** [comment]
Related:
- https://github.com/statelyai/xstate/issues/3854
- https://github.com/statelyai/xstate/issues/3845

**Reply by Andarist** [reply]
I think that @zhangjialu is right here:
> It seems that typegen doesn't know the actions returned by pure action

I don't think that we extract those from `pure` - the whole `pure` action might be handled poorly by the extractor right now.
